src/Parentheses.cpp

```diff
@@ -1,5 +1,29 @@
 #include "Parentheses.hpp"
+#include "stack"
 
 bool Parentheses::isBalanced(std::string s) {
-    return true;
-}
\ No newline at end of file
+
+    std::stack<char> stos{};
+    
+    for(auto el:s) {
+        if(stos.size() >= 1) {
+            if ((el == ')') and (stos.top() == '(')) {
+                    stos.pop();
+            }
+            else if ((el == '}') and (stos.top() == '{')) {
+                    stos.pop();
+            }
+            else if ((el == ']') and (stos.top() == '[')) {
```

Comment:
pomyślcie, czy dałoby się to jeszcze zrefaktoryzować, aby zasada Open Close tutaj była spełniona :)
Póki co jeśli by doszła nowa para nawiasów np <> to trzeba by było nowego ifa dopisać, czyli zmodyfikować ten algorytm. A jak można napisać taki algorytm, aby go potem nie modyfikować jeśli mamy nowy przypadek?

Reply:
Czy refactor jaki zaproponowalam (przeniesienie funkcjonalnosci sprawdzajacej czy jest para dla nawiasu poza algorytm) jest wystarczajacy?